Editor/CoreAPI/StateManager.cs

```diff
@@ -281,6 +280,7 @@ public void RefreshProfiles()
         public PutSettingResponse SaveProfiles()
         {
             var profiles = _serializer.Serialize(_allProfiles);
+            OnUserProfileUpdated?.Invoke();
```

Comment:
This has caused regression. When I try to deploy a managed fleet, the BuildFilePath and BuildFolderPath are null because of this change, and the change made at line 86. I am reverting these two lines in a separate change, please take a second look to see what the right fix here should be.